presto-mongodb/src/main/java/io/prestosql/plugin/mongodb/MongoSession.java

```diff
@@ -407,6 +409,14 @@ else if (singleValues.size() > 1) {
             return Optional.of(prestoNativeValue);
         }
 
+        if (type == DATE) {
+            return Optional.of(new Date(TimeUnit.DAYS.toMillis((Long) prestoNativeValue)));
+        }
+
+        if (type == TIMESTAMP_MILLIS) {
+            return Optional.of(new Date(TimeUnit.MILLISECONDS.convert((Long) prestoNativeValue, TimeUnit.MICROSECONDS)));
+        }
```

Comment:
Can we add support for `timestamp with time zone` as well?

Reply:
@ebyhr yep, updated to add that.